src/Microsoft.DotNet.Cli.Utils/MSBuildForwardingAppWithoutLogging.cs

```diff
@@ -78,20 +81,20 @@ private static string GetMSBuildSDKsPath()
                 SdksDirectoryName);
         }
 
-        private static string GetRunVbcPath()
+        private static string GetRunToolPath()
         {
-            return GetRunToolPath("Vbc");
-        }        
+            return Path.Combine(AppContext.BaseDirectory, "Roslyn", "bincore");
+        }
 
-        private static string GetRunCscPath()
+        private static string GetRunToolExe(string compilerName)
         {
-            return GetRunToolPath("Csc");
+            var scriptExtension = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".cmd" : "";
+            return $"Run{compilerName}{scriptExtension}";
         }
 
-        private static string GetRunToolPath(string compilerName)
+        private static string GetDotnetPath()
         {
-            var scriptExtension = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".cmd" : ".sh";
-            return Path.Combine(AppContext.BaseDirectory, "Roslyn", $"Run{compilerName}{scriptExtension}");
+            return Process.GetCurrentProcess().MainModule.FileName;
```

Comment:
There is an API for this called `new Muxer().MuxerPath`.  We should be consistent in how we find the `dotnet` executable.